tests/Integration/Behaviour/Features/Context/Domain/Discount/DiscountFeatureContext.php

```diff
@@ -268,6 +305,18 @@ protected function createDiscount(string $cartRuleReference, array $data, AddDis
             }
         }
 
+        if ($command instanceof AddFreeGiftDiscountCommand) {
+            if (!empty($data['gift_product'])) {
+                $reference = $data['gift_product'];
+                $productId = $this->getProductIdByReference($reference);
+                $command->setProductId((int) $productId);
+            }
+
+            if (!empty($data['gift_combination'])) {
+                $command->setCombinationId((int) $data['gift_combination']);
```

Comment:
```suggestion
                $command->setCombinationId($this->referenceToId((int) $data['gift_combination']));
```
The combination should also be based on references